planet/scripts/cli.py

```diff
@@ -103,3 +107,37 @@ def init(email, password):
     response = call_and_wrap(clientv1().login, email, password)
     write_planet_json({'key': response['api_key']})
     click.echo('initialized')
+
+@cli.command('quota')
+def quota():
+    '''Print allocation and remaining quota in Sqkm.'''
+    try:
+        with open (os.path.join(os.path.expanduser('~'), '.planet.json')) as f:
+            data=json.load(f)
+        main=requests.get('https://api.planet.com/auth/v1/experimental/public/my/subscriptions', auth=HTTPBasicAuth(data["key"], ''))
+        if main.status_code==200:
+            content=main.json()
+            n=(str(main.json()).count("quota_sqkm"))
+            i=0
+            while i<n:
+                item_id=(content[i])
+                print(" ")
+                print("Allocation Name: "+str(item_id['organization']['name']))
+                print("Allocation active from: "+str(item_id['active_from']).split("T")[0])
+                print("Quota Enabled: "+str(item_id['quota_enabled']))
+                print("Total Quota in SqKm: "+str(item_id['quota_sqkm']))
+                print("Total Quota used: "+str(item_id['quota_used']))
+                if (item_id['quota_sqkm'])is not None:
+                    print("Remaining Quota in SqKm: "+str(float(item_id['quota_sqkm'])-float(item_id['quota_used'])))
+                else:
+                    print("No Quota Allocated")
+                print("")
+                i=i+1
+        else:
+            print("Failed with exception code: "+str(main.status_code))      
+    except Exception as e:
```

Comment:
Using `except Exception` is a bad idea! What happens if anything in the block above throws an Exception that's _not_ related to an un-initialized client (e.g., what if the response JSON is missing an expected field, and one of the `print` lines throws a `KeyError`?)? We'll never know about it, and the CLI will just try to ask the user to initialize regardless.

Instead, do:
`except IOError:  # note that you don't need "as e"`

Reply:
Resolved